ui/js/controllers/perf/e10s-trend.js

```diff
@@ -0,0 +1,568 @@
+            $scope.newResults = {};
+            $scope.titles = {};
+
+            // ** get dates for original base date (for the last x days back from the base date) **
+            // selectedBaseDate and selectedNewDate are in SECONDS from today back i.e. 1 day ago, 1 week ago
+
+            // calculate the actual base date itself, as a date not just x seconds back from today
+            var today_epoch = Date.now();
+            var base_date = today_epoch - ($scope.selectedBaseDate.value * 1000)  // want MS
+
+            // start date for retrieving data is the base date minus sample size i.e. x days back from base date
+            var data_start_date = base_date - ($scope.selectedSampleSize.value * 1000)  // want MS
+            data_start_date = new Date(data_start_date);
+            // must be in format 'YYYY-MM-DDT:HH:MM:SS' ie. 2016-09-10T1:1:1
+            $scope.base_data_start_date = data_start_date.toISOString().slice(0, -5);
+
+            // end date for retrieving data is the base date itself
+            // must be in format 'YYYY-MM-DDT:HH:MM:SS' ie. 2016-09-10T1:1:1
+            var data_end_date =  new Date(base_date);
+            $scope.base_data_end_date = data_end_date.toISOString().slice(0, -5);
+
+            // ** get dates for new date (for the last x days back from the new date) **
+            // calculate the actual new date itself, as a date not just x seconds back from today
+            var today_epoch = Date.now();
+            var new_date = today_epoch - ($scope.selectedNewDate.value * 1000)  // want MS
+
+            // start date for retrieving data is the new date minus sample size i.e. x days back from new date
+            var data_start_date = new_date - ($scope.selectedSampleSize.value * 1000)  // want MS
+            data_start_date = new Date(data_start_date);
+            // must be in format 'YYYY-MM-DDT:HH:MM:SS' ie. 2016-09-10T1:1:1
+            $scope.new_data_start_date = data_start_date.toISOString().slice(0, -5);
+
+            // end date for retrieving data is the new date itself
+            // must be in format 'YYYY-MM-DDT:HH:MM:SS' ie. 2016-09-10T1:1:1
+            var data_end_date =  new Date(new_date);
+            $scope.new_data_end_date = data_end_date.toISOString().slice(0, -5);
+
+            PhSeries.getSeriesList(
+                $scope.selectedRepo.name,
+                { start_date: $scope.base_data_start_date,
+                  end_date: $scope.base_data_end_date,
+                  subtests: 0,
+                  framework: 1
+                }).then(
+                    function(originalSeriesList) {
+                        return _.filter(originalSeriesList, function(series) {
+                            return series.options.indexOf('pgo') >= 0 ||
+                                (series.platform === 'osx-10-10' &&
+                                 series.options.indexOf('opt') >= 0);
+                        });
+                    }).then(
```

Comment:
I'm not sure why there's a then() here, as the previous block of code doesn't operate asynchronously. You should be able to just filter the originalSeriesList and continue, I think?